### PR TITLE
Modify time profiler

### PR DESCRIPTION
**Pull request by davidegorbani**
@S-Dafarra modified the [TimeProfiler](https://github.com/robotology/walking-controllers/blob/56071b8ca640800667bb81b185c58307c7303ca7/src/TimeProfiler/src/TimeProfiler.cpp) to use the `std::chrono` to get the time. Moreover, he added a check on the time required to get the feedback in the [WalkingModule](https://github.com/robotology/walking-controllers/blob/56071b8ca640800667bb81b185c58307c7303ca7/src/WalkingModule/src/Module.cpp).

### CODE REVIEW

File: src/TimeProfiler/include/WalkingControllers/TimeProfiler/TimeProfiler.h [hunk 1]
```diff
@@ -21,8 +22,8 @@ namespace WalkingControllers
      */
     class Timer
     {
-        clock_t m_initTime; /**< Init time. */
-        clock_t m_endTime; /**< End time. */
+        std::chrono::time_point<std::chrono::system_clock> m_initTime; /**< Init time. */
+        std::chrono::time_point<std::chrono::system_clock> m_endTime; /**< End time. */
```
Comment (traversaro) [inline src/TimeProfiler/include/WalkingControllers/TimeProfiler/TimeProfiler.h:26]:
```suggestion
        std::chrono::time_point<std::chrono::steady_clock> m_initTime; /**< Init time. */
        std::chrono::time_point<std::chrono::steady_clock> m_endTime; /**< End time. */
```

File: src/TimeProfiler/src/TimeProfiler.cpp [hunk 2]
```diff
@@ -26,18 +26,18 @@ void Timer::resetAverageDuration()
 
 void Timer::setInitTime()
 {
-    m_initTime = clock();
+    m_initTime = std::chrono::system_clock::now();
```
Comment (traversaro) [inline src/TimeProfiler/src/TimeProfiler.cpp:29]:
```suggestion
    m_initTime = std::chrono::steady_clock::now();
```

File: src/TimeProfiler/src/TimeProfiler.cpp [hunk 3]
```diff
@@ -26,18 +26,18 @@ void Timer::resetAverageDuration()
...
 }
 
 void Timer::setEndTime()
 {
-    m_endTime = clock();
+    m_endTime = std::chrono::system_clock::now();
```
Comment (traversaro) [inline src/TimeProfiler/src/TimeProfiler.cpp:34]:
```suggestion
    m_endTime = std::chrono::steady_clock::now();
```

**Review by traversaro — APPROVED** [review]
Minor comment: `std::chrono::system_clock` may be non monotonic (see https://en.cppreference.com/w/cpp/chrono/system_clock), i.e. it can go back in time, for example if there is NTP or PTP synchronization of if the user change the system time. A better alternative may be  `std::chrono::steady_clock`, that instead is always monotonic.

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**